stats/cloud/collector.go

```diff
@@ -441,15 +443,28 @@ func (c *Collector) flushHTTPTrails() {
 	c.aggrBuckets = map[int64]aggregationBucket{}
 	c.bufferSamples = append(c.bufferSamples, newSamples...)
 }
+
+func (c *Collector) shouldStopSendingMetrics(err error) bool {
+	if err == nil {
+		return false
+	}
+
+	return strings.HasPrefix(err.Error(), "(403/E4)")
+}
+
 func (c *Collector) pushMetrics() {
-	c.bufferMutex.Lock()
+	c.mu.Lock()
+	if c.stopSendingMetrics {
```

Comment:
I think this is just ... wrong. IMO you should not get to pushMetrics if we are no longer going to send metrics. 
You should 
1. stop the cycle that calls pushMetrics
2. stop collecting metrics in Collect 
Otherwise you are still doing parts of the work for no reason

Reply:
Ah, ok, maybe I mis-understood the purpose description in the issue

Reply:
I agree with @MStoykov, once we decide that we won't send any metrics to the cloud, then `Collect()` should just become a no-op and we should probably just return from `Run()`, i.e. stop all of the data crunching and data pushing timers and stop the collector from doing any work.

Reply:
@na-- @MStoykov Agree, I mis-understood the requirement from scratch, will push new patch then.